### PR TITLE
Adding a Switch for BGRA mode

### PR DESCRIPTION
**Pull request by GoBig87**
Addding code to switch from BGRA to RGBA if BGRA is detected.  IOS uses BGRA for its camera but PIL doesn't support it.  Switching the RGBA for PIL will just make the colors wrong wrong on analysis but for detecting QR codes its fine.  The camera inside the app still presents the right colors to users.

### CODE REVIEW

**Comment by AndreMiras** [comment]
Nice thanks! Do you mind fixing the linter issue raised by Travis?
You can run test locally with `make test` or just linting with `tox -e pep8`

**Comment by GoBig87** [comment]
I've never used Travis.  I'll read up on it and get the code formated right so the issues go away.

**Comment by GoBig87** [comment]
I fixed the spaces after the comments and it said everything passed.

File: zbarcam/zbarcam.py [hunk 1]
```diff
@@ -79,6 +79,11 @@ def _detect_qrcode_frame(self, instance, camera, texture):
         image_data = texture.pixels
         size = texture.size
         fmt = texture.colorfmt.upper()
+        # PIL doesn't support BGRA but IOS uses BGRA for the camera
+        # if BGRA is detected it will switch to RGBA, color will be off
+        # but we don't care as it's just looking for barcodes
+        if fmt == 'BGRA':
```
Comment (AndreMiras) [inline zbarcam/zbarcam.py]: Should we `if platform == 'ios' and fmt == 'BGRA':` so we're sure we don't break things for other platforms?
What if for example `RGBA` is not supported in one of the other platforms?

**Comment by AndreMiras** [comment]
Thanks for the linting fix @GoBig87 I have a last concert regarding the `RGBA` on other platform. I left a comment, tell me what you think :pray:

**Comment by GoBig87** [comment]
The error comes from PIL.images.frombytes where https://pillow.readthedocs.io/en/3.3.x/handbook/concepts.html#concept-modes.  BGRA isn't a supported mode. So it shouldn't be platform specific.  I dont know enough about image formats to know if there are any other major formats that could break zbar if the format wasnt supported by PIL.  Maybe a better fix would be a way to convert all images to a known format?  Well for now it works this might be more of a band aid than a proper fix.  If more issues pop up of unsupported modes for PIL maybe a better solution will have to be developed.

**Comment by AndreMiras** [comment]
But my point is more that, I know it already works as it is on Linux and Android. But I don't know if your change could introduce a regression on these two platforms. So since it's a hack to make it work on iOS, I though maybe it makes sense to only activate the hack `if platform == 'ios' and fmt == 'BGRA'`. Do you see my point?

**Comment by GoBig87** [comment]
I see your point.  I added a switch to only switch for ios users.  If you haven't had anyone complain about PIL crashing in zbarcam it might be kind of rare outside of ios for devices to user BGRA.

**Pull request merged**